src/language_concepts/data_types.md

```diff
@@ -1,206 +1,192 @@
-extra range constraints and so is less efficient rather than more.
-
-```rust,noplaypen
-fn main(x : Field, y : Field) {
-    let z = x + y;
-}
-```
+Public data types are treated no differently to private types apart from the fact that their values will be revealed in proofs generated. Simply changing the value of a public type will not change the circuit (where the same goes for changing values of private types as well).
 
-`x`, `y` and `z` are witness types (private `Field`). Using the `let` keyword we derived a new witness `z` which is constrained to be equal to `x + y`.
+_Private_ are also referred to as _witnesses_ sometimes.
 
-#### Integer Type
+> **Note:** The terms private and public when applied to a type (e.g. `pub Field`) have a different meaning than when applied to a function (e.g. `pub fn foo() {}`).
+>
+> The latter is a visibility modifier familiar to those coming from other languages, while the former can be thought of as a visibility modifier for the Verifier to interpret.
 
-An integer type is a witness type which has been constrained using a range constraint.
-The Noir front-end currently supports arbitrary sized integer types.
+### pub Modifier
 
-Below we show the integer type in action:
+All data types in Noir are private by default. Types are explicitly declared as public using the `pub` modifier:
 
 ```rust,noplaypen
-fn main(x : Field) {
-    let y = x as u24;
+fn main(x : Field, y : pub Field) -> pub Field {
+    x + y
 }
 ```
 
-`x` and `y` are both private types (witnesses), however `y` is an integer type.
-If `y` exceeds the range \\([0,2^{24}-1]\\) then any proof created will output false by the verifier.
+In this example, the first input (i.e. `x`) is private while the second input (i.e. `y`) and the return value (i.e. `x + y`) are public.
 
-> **Note:** The Aztec backend supports even and odd sized integer types, so while using the Aztec backend, even sized integer types such as u32, u48, as well as odd sized integer types such as u5, u3 will produce proofs.
+> **Note:** As of this moment, public types can only be declared through parameters on `main`.
 
-#### Constants
-
-A constant type is a value that does not change per circuit instance. This is different to a witness which changes per proof.
-If a constant type that is being used in your program is changed, then your circuit will also change.
+## Primitive Types
 
-Below we show how to declare a constant value:
+A primitive type represents a single value. They can be private or public.
 
-```rust,noplaypen
-fn main() {
-    let a: comptime Field = 5;
+### The Field Type
 
-    // `comptime Field` can also be inferred:
-    let a = 5;
-}
-```
+The `Field` type corresponds to the native field type of the proving backend. For example, this would be a (roughly) 256-bit integer when using the default TurboPlonk backend.
 
-Note that variables declared as mutable may not be constants:
+Fields are simply defined with the keyword `Field`:
```

Comment:
Fields are not defined with the Field keyword, they are defined with `let` statements like other values. I think we can remove this sentence. Perhaps replacing it with a brief explanation of ways in which we can use fields. E.g. "Fields support integer arithmetic and are often used as the default numeric type in Noir" or similar.